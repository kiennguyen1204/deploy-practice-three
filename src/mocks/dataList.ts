export const dataItemDetail = {
  effects: [
    {
      title: 'effects',
      image: '/src/assets/images/products/color_filter.png',
      content: 'Calming, Creative, Happy, Relaxing, Sleepy, Uplifting'
    },
    {
      title: 'relieve',
      image: '/src/assets/images/products/brain.png',
      content:
        'Anxiety, Arthritis, Chronic Pain, Depression, Fatigue, Inflammation, Insomnia, Irregular Bowel Movements, Migraines, Mood Swings'
    },
    {
      title: 'aromas',
      image: '/src/assets/images/products/carbon_smoke.svg',
      content: 'Chemical, Citrus, Earthy, Pungent, Sour'
    }
  ]
};

export const DATA_LIST = [
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  },
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  },
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  },
  {
    isSoldOut: 86,
    title: '',
    totalReviews: 135,
    discount: '$200.00',
    price: '$102.00',
    unit: 'unit 1',
    volume: [],
    image: 'https://tinyurl.com/2s34y377',
    sale: 'Indica 70%',
    description:
      'Jungle Diamonds is a slightly indica dominant hybrid strain (60% indica/40% sativa) created through crossing the infamous Slurricane X Gorilla Glue #4 strains.',
    category: 'CONCENTRATES',
    name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
    rate: '46/5',
    id: '1'
  }
];

export const LIST_PRODUCT_CATEGORY_FILTER = [
  {
    name: 'Sales',
    value: 12
  },
  {
    name: 'Cannabis',
    value: 430
  },
  {
    name: 'Pre-Rolls',
    value: 40
  },
  {
    name: 'CBD Oil',
    value: 12
  },
  {
    name: 'Magic Mushrooms',
    value: 34
  },
  {
    name: 'Extracts',
    value: 26
  },
  {
    name: 'Edibles',
    value: 32
  },
  {
    name: 'Vape Pens',
    value: 12
  },
  {
    name: 'Accessories',
    value: 10
  },
  {
    name: 'Bath & Body',
    value: 8
  },
  {
    name: 'Bundles',
    value: 24
  },
  {
    name: 'Wholesale',
    value: 28
  }
];
