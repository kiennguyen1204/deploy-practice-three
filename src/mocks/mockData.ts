import { Product } from 'interfaces/item';

export const DATA_ITEM = {
  image: 'test-image.jpg',
  subtitle: 'Test Subtitle',
  title: 'Test Title',
  description: 'Test Description',
  to: '/test-link'
};

export const MOCK_PROPS = {
  image: '/sample_image.png',
  title: 'Sample Title',
  description: 'Sample Description',
  to: '/sample-link',
  shop: 'Sample Shop',
  icon: '/sample_icon.png'
};

export const MOCK_ITEM: Product = {
  id: 'sampleId',
  isSoldOut: true,
  discount: 20,
  image: '/src/assets/images/default_image.webp',
  title: 'Sample Title',
  category: 'Sample Category',
  name: 'Sample Name',
  rate: '4.5',
  totalReviews: 25,
  sale: 'Sale!',
  price: 8,
  volume: ['Volume 1', 'Volume 2'],
  description: 'Sample Description',
  reviews: 0
};

export const MOCK_DATA_CUSTOMER = {
  userName: 'Vikki Starr',
  avatar: '/src/assets/images/reviews/VikkiStarr.png',
  comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
  star: 4,
  date: 'January 15, 2023'
};

export const dataCart = {
  total: 305.0,
  shipping: 50,
  payTotal: 355,
  discount: 0,
  cart: [
    {
      name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
      image: '/src/assets/images/products/rind-remix-pina-colada-snack-mix-front_2000x.webp',
      price: 120.0,
      amount: 2,
      total: 240.0,
      integra: [
        {
          namePack: 'Add Integra Pack - 4g',
          amount: 1,
          price: 2.0,
          total: 2.0
        },
        {
          namePack: 'Add Integra Pack - 8g',
          amount: 1,
          price: 3.0,
          total: 3.0
        }
      ],
      subTotal: 245.0
    },
    {
      name: 'Mix And Match Shatter/Budder 28g (4 X 7G)',
      image: '/src/assets/images/products/rind-snacks-chewy-straw-peary-front_2000x.webp',
      price: 13.0,
      amount: 5,
      total: 65.0,
      integra: [],
      subTotal: 65.0
    }
  ]
};
