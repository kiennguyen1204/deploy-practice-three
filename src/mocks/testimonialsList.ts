export const TESTIMONIALS_LIST = [
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 4,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 4,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  },
  {
    userName: 'Vikki Starr',
    avatar: '/src/assets/images/reviews/VikkiStarr.png',
    comment: `Absolutely love TopShelfBC; affordable on any budget and such fast delivery, straight to my door! I recommend them to all my friends and family for their 420 needs.`,
    star: 5,
    date: 'January 15, 2023'
  }
];
