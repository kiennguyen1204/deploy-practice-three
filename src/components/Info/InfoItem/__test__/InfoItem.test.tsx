import { render, screen } from '@testing-library/react';

// mocks
import { MOCK_PROPS } from 'mocks/mockData';

// components
import BoxInfoItem from 'components/Info/InfoItem';
import { MemoryRouter } from 'react-router-dom';

describe('BoxInfoItem', () => {
  it('BoxInfoItem renders with the correct content and attributes', () => {
    // 'to' is wrapped in a <NavLink> so it should be wrapped in a MemoryRouter
    const { asFragment } = render(
      <MemoryRouter>
        <BoxInfoItem {...MOCK_PROPS} />
      </MemoryRouter>
    );

    // Check if the image elements have the correct 'src' and 'alt' attributes
    const imageElement = screen.getByAltText(MOCK_PROPS.title);
    expect(imageElement).toHaveAttribute('src', MOCK_PROPS.image);
    expect(imageElement).toHaveAttribute('alt', MOCK_PROPS.title);

    expect(asFragment()).toMatchSnapshot();
  });
});
