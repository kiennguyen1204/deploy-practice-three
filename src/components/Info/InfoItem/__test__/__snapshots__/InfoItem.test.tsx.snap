// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxInfoItem BoxInfoItem renders with the correct content and attributes 1`] = `
<DocumentFragment>
  <section
    class="item-box"
    data-testid="box-info-item"
  >
    <picture
      class="flex-row-container card-image"
    >
      <img
        alt="Sample Title"
        src="/sample_image.png"
      />
      <img
        alt="Sample Title Icon"
        aria-hidden="true"
        class="flex-container icon-image"
        src="/sample_icon.png"
      />
    </picture>
    <h2
      class="title"
      data-testid="title"
    >
      Sample Title
    </h2>
    <p
      class="description"
      data-testid="description"
    >
      Sample Description
    </p>
    <a
      aria-label="Visit Sample Shop"
      class="shop"
      data-testid="shop"
      href="/sample-link"
    >
      Sample Shop
    </a>
  </section>
</DocumentFragment>
`;
